### DISCUSSION TITLE
Closing a TLS socket session without closing the underlying socket

### DISCUSSION THREAD

**Discussion by lppedd** [nodejs/help#4637] [Q&A]
As per the title, currently closing the TLS socket client also closes the underlying plain socket. This is sometimes unwanted behavior.

I've encountered this problem while translating the following Java code:

```java
SSLContext ctx = ...
ctx.getSocketFactory().createSocket(
  socket, 
  hostName, 
  portNumber, 
  false /* Don't close the underlying socket when this socket is closed */
);
```

Any idea how I could replicate the same behavior? I'm open to using non-public (but still accessible) functions.

**Comment by fcastrocs** [comment]
Did you get anywhere with this?

I am having the same issue. Using a netSocket to connect to a proxy, then passing it to tls.connect(). There should be a way to have the underlying netSocket remain open if tlsSocket drops. I tested, even tlsSocket.end() and tlsSocket.destroy() also close the netSocket.

The following statement from the docs is false.

"Connection/disconnection/destruction of socket is the user's responsibility"

https://nodejs.org/api/tls.html#tlsconnectoptions-callback

**Reply by lppedd** [reply]
Hey. No I didn't get anywhere. In the end we had to re-engineer the backend specifically for this issue. Very unfortunate.
